website/docs/language/import/index.mdx

```diff
@@ -0,0 +1,188 @@
+---
+page_title: Import - Configuration Language
+description: >-
+  Import resources into Terraform to bring them under Terraform's management by writing `import` blocks in code.
+---
+
+# Import
+
+-> **Note:** Import blocks are only available in Terraform v1.5.0 and later.
+
+~> **Experimental:** Import blocks are an experimental Terraform feature. While we do not expect to make backwards-incompatible changes to their syntax, processing of imports during the plan, generating configuration, and the `-generate-config-out` flag may change in future releases.
+
+Terraform can import existing infrastructure resources, which brings them under Terraform's management. Once you have imported a resource, it will be present in the Terraform configuration and state, just like a resource you had created via Terraform. You can then manage it like any other resource, updating its attributes and destroying it as part of a normal resource lifecycle.
+An `import` block records that Terraform imported the resource, that it was not originally created due to this Terraform configuration. Considered another way, it records the user's intent that an import be added to Terraform's plan graph.
+
+## `import` block syntax
+
+The `import` block can appear in any Terraform configuration file. A common pattern is to create an `imports.tf` file, or to place each `import` block next to the `resource` block it will import into.
+
+```hcl
+import {
+  to = aws_instance.example
+  id = "i-abcd1234"
+}
+```
+
+The above `import` block defines an import of the AWS instance with ID "i-abcd1234" into the resource address `aws_instance.example` in the root module.
```

Comment:
Terraform really pushes the limits of active voice docs with its declarative language and implicit, graph-based operation.

Declaring a `resource` in config, for example, is not telling Terraform to create that resource, it is telling Terraform that your desired state-of-the-world is one in which that resource exists just so. If during the plan stage Terraform determines that it must create the resource in order for this to be true, it plans a create (n.) and during the apply stage creates (v.) the resource.

### Long meditation on configuration-driven behaviour

Configuration-driven move stretched this metaphor a little, as the `moved` block on its own is more concerned with Terraform's internal state than the state-of-the-world. Declaring a `moved` block in config is telling Terraform that the desired state is one in which a resource that used to be at the `from` address is now at the `to` address. If that still needs to happen, Terraform plans a move and then moves the resource.
Does a `moved` block "move" a resource? In my opinion, no--especially since if the resource is already at the new address, Terraform's graph walk outputs a no-op action instead of move. Terraform moves resources, but you can [Use Configuration to Move Resources](https://developer.hashicorp.com/terraform/tutorials/configuration-language/move-config). 

The `import` block has some similarities to the move case, but involves both state and state-of-the-world. Use Configuration to Import Resources, by writing an `import` block in config, which declares your intent that the targeted resource be understood to have been created outside Terraform and therefore the subject of an import (n.). This stilted and indirect technical language is very elegantly translated into understandable prose in your import tutorial (just skimmed the PR).

[As you can see, there is some inconsistency over whether the past participle (`moved`) or noun-that-can-be-mistaken-for-a-verb (`import`) is more appropriate here, which I would like to resolve in future releases. "The `import` block" just proved too compelling to force the issue now.]

### Conclusion

In terms of user intent, you declare in config what you want to be the case, and at worst what you want to have happened _at some point up to and including the next apply_. Only the graph's inscrutable interior, not the practitioner, is a valid source of an intent to perform any given action in particular.